Hash_map/benchmark/Hash_map/polyhedron.cpp

```diff
@@ -18,125 +16,134 @@
 #include <boost/random/random_number_generator.hpp>
 #include <boost/random/linear_congruential.hpp>
 
-typedef CGAL::Simple_cartesian<double>  Kernel;
-typedef Kernel::Point_3                 Point_3;
+typedef CGAL::Simple_cartesian<int>  Kernel;
+struct Point_3 : Kernel::Point_3 {
+  using Kernel::Point_3::operator=;
+  Point_3() : Kernel::Point_3(0,0,0) {}
+  Point_3(int i) : Kernel::Point_3(i,0,0) {}
+};
 typedef Kernel::Vector_3                Vector_3;
 
 
 typedef CGAL::Polyhedron_3<Kernel>      Mesh;
+typedef boost::property_map<Mesh,CGAL::vertex_point_t>::type  VPM;
 typedef boost::graph_traits<Mesh>::vertex_descriptor vertex_descriptor;
+typedef std::vector<vertex_descriptor> Vertex_list;
 
 typedef CGAL::Timer                     Timer;
 
+template <typename Map>
+auto reserve(Map& sm, typename Map::size_type ii) -> decltype(sm.reserve(ii), void()){
+    sm.reserve(ii);
+}
 
 
-void  fct(int ii, int jj)
-{
-  typedef std::map<vertex_descriptor,Point_3> SM;
-  typedef std::unordered_map<vertex_descriptor,Point_3> SUM;
-  typedef boost::unordered_map<vertex_descriptor,Point_3> BUM;
-  typedef CGAL::Unique_hash_map<vertex_descriptor, Point_3> UHM;
-
+template <typename Map>
+Point_3 lookup(Map& sm,vertex_descriptor vh){
+    auto search = sm.find(vh);
+    if(search != sm.end())
+        return search->second;
+    return Point_3();
+}
 
-  Mesh mesh;
-  typedef boost::property_map<Mesh,CGAL::vertex_point_t>::type  VPM;
-  VPM vpm = get(CGAL::vertex_point,mesh);
+template <typename X, typename Y>
+Point_3 lookup(CGAL::Unique_hash_map<X,Y>& sm,vertex_descriptor vh){
+    const CGAL::Unique_hash_map<X,Y>& const_sm = sm;
+    return const_sm[vh];
+}
 
+template <typename Map>
+double fct(int ii, int jj, const Vertex_list& V, const Vertex_list& V2, const VPM& vpm, const std::string& s)
```

Comment:
I don't know why `fct` returns a `double`, when `int x` and `int res`/`int temp`

Reply:
Should be just int.
I do this adding to make sur that the compiler does not optimize loops away that do nothing.
And also to check that for all the map types the result is the same. 

Btw, we should maybe only work in your branch GilesBathgate/Hash_map-benchmark-GilesBathgate 
instead of this one.